core/variant_op.cpp

```diff
@@ -161,35 +162,78 @@ bool Variant::booleanize(bool &r_valid) const {
 			return _data._real;
 		case STRING:
 			return (*reinterpret_cast<const String *>(_data._mem)) != "";
-		case VECTOR2:
-		case RECT2:
-		case TRANSFORM2D:
-		case VECTOR3:
-		case PLANE:
-		case RECT3:
-		case QUAT:
-		case BASIS:
-		case TRANSFORM:
-		case COLOR:
+		case VECTOR2: {
+			const Vector2 *v = reinterpret_cast<const Vector2 *>(_data._mem);
+			return v->x + v->y;
+		}
+		case RECT2: {
+			const Rect2 *v = reinterpret_cast<const Rect2 *>(_data._mem);
+			return v->position.x + v->position.y + v->size.x + v->size.y;
+		}
+		case TRANSFORM2D: {
+			const Transform2D *v = _data._transform2d;
+			return v->elements[0].x + v->elements[0].y +
```

Comment:
Same here, should be false if all components are null. I guess you could do `booleanize(elements[0]) || booleanize(elements[1]) || booleanize(elements[2])`, though maybe that would be less efficient than just hardcoding the six checks :)